beacon-chain/blockchain/init_sync_process_block.go

```diff
@@ -8,11 +8,20 @@ import (
 	"github.com/prysmaticlabs/prysm/consensus-types/wrapper"
 )
 
-// This saves a beacon block to the initial sync blocks cache.
-func (s *Service) saveInitSyncBlock(r [32]byte, b interfaces.SignedBeaconBlock) {
+// This saves a beacon block to the initial sync blocks cache. It rate limits how many blocks
+// the cache keeps in memory (2 epochs worth of blocks) and saves them to DB when it hits this limit.
+func (s *Service) saveInitSyncBlock(ctx context.Context, r [32]byte, b interfaces.SignedBeaconBlock) error {
```

Comment:
Because every single call to this function came with a direct call afterward to prune, since we want to keep the cache to a fixed size, then here's the best point to prune immediately. But I don't feel strongly about this, I can revert back to the previous behavior.